src/vanilla.ts

```diff
@@ -16,7 +17,9 @@ export const ref = <T extends object>(o: T): T & AsRef => {
 const isSupportedObject = (x: unknown): x is object =>
   typeof x === 'object' &&
   x !== null &&
-  (Array.isArray(x) || !(x as any)[Symbol.iterator]) &&
+  (Array.isArray(x) ||
+    (x as any)[CUSTOM_TYPE] ||
+    !(x as any)[Symbol.iterator]) &&
```

Comment:
Given that the `proxySet` implements the iterator symbol I needed a way to bypass this check. I didn't want to expose something from `utils` inside the `core` package. So I thought that having a dedicated symbol to identity custom data types would be useful. 

It would be better to move the symbol in a dedicated file, maybe.

If you see a better way to solve this issue, I'm open to it 😊

Reply:
As you might guess, I don't prefer this solution. tbh, I don't like how `isSupportedObject` is hacked by me. Give me some time for this. I'd like to come up with an idea I'm happy with.

Reply:
Fully agree with you on this one. Can't we replace `!(x as any)[Symbol.iterator])` by explicit check for `Map` and `Set`?  If I'm not mistaken those are the built-in objects that implement the iterator: `String`, `Array`, `TypedArray`, `Map`, and `Set`

```js
!(x instanceof Map) &&
!(x instanceof Set) &&
!(x instanceof TypedArray)
```

Reply:
I will work on the vanilla part soon (not strictly tied to this PR). Please review it later to see if it works for this PR.